File: backend/src/main/java/ma/investpro/dto/FournisseurDTO.java
package ma.investpro.dto;

import jakarta.validation.constraints.*;
import lombok.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FournisseurDTO {
    private Long id;

    @NotBlank(message = "Le code est obligatoire")
    @Size(max = 50)
    private String code;

    @NotBlank(message = "La raison sociale est obligatoire")
    @Size(max = 200)
    private String raisonSociale;

    @Size(max = 20)
    @Pattern(regexp = "^[0-9]*$", message = "L'IF doit contenir uniquement des chiffres")
    private String identifiantFiscal;

    @Size(min = 15, max = 15, message = "L'ICE doit contenir 15 caractères")
    @Pattern(regexp = "^[0-9]{15}$", message = "L'ICE doit contenir 15 chiffres")
    private String ice;

    private String adresse;

    @Size(max = 100)
    private String ville;

    @Pattern(regexp = "^[0-9+\\-\\s()]*$")
    private String telephone;

    @Pattern(regexp = "^[0-9+\\-\\s()]*$")
    private String fax;

    @Email
    private String email;

    @Size(max = 100)
    private String contact;

    private Boolean nonResident;
    private String remarques;
    private Boolean actif;
}

File: backend/src/main/kotlin/ma/investpro/dto/FournisseurDTO.kt
package ma.investpro.dto

import java.time.LocalDateTime

/**
 * DTO pour Fournisseur - Entité simple sans relations circulaires
 */
data class FournisseurDTO(
    val id: Long?,
    val code: String,
    val raisonSociale: String,
    val identifiantFiscal: String?,
    val ice: String?,
    val adresse: String?,
    val ville: String?,
    val telephone: String?,
    val fax: String?,
    val email: String?,
    val contact: String?,
    val nonResident: Boolean,
    val remarques: String?,

    val createdAt: LocalDateTime?,
    val updatedAt: LocalDateTime?,
    val createdBy: String?,
    val updatedBy: String?,
    val actif: Boolean
)

/**
 * DTO simplifié pour Fournisseur - Utilisé dans les listes
 */
data class FournisseurSimpleDTO(
    val id: Long?,
    val code: String,
    val raisonSociale: String,
    val ice: String?,
    val actif: Boolean
)
